### PR TITLE
[FREELDR] Implement the FASTDETECT option and enable it by default

### PR DESCRIPTION
**Pull request by disean**
## Purpose

The serial mouse is recognized by the serial stack since NT5.x.

### CODE REVIEW

File: boot/freeldr/freeldr/arch/arm/macharm.c
Comment (SergeGautherie) [inline boot/freeldr/freeldr/arch/arm/macharm.c:97]: Add `UNREFERENCED_PARAMETER()`?

File: boot/freeldr/freeldr/arch/i386/pc/machpc.c
Comment (SergeGautherie) [inline boot/freeldr/freeldr/arch/i386/pc/machpc.c:740]:
```suggestion
static
ULONG
```

**Comment by HBelusca** [comment]
Note: see https://github.com/HBelusca/reactos/commit/93d6b9accf1b5f76892e8828befa350615893d61

File: boot/freeldr/freeldr/arch/i386/pc/machpc.c
Comment (HBelusca) [inline boot/freeldr/freeldr/arch/i386/pc/machpc.c:749]: ah, this syntax with the colon is also supported on Windows? interesting...

**Review by HBelusca — COMMENTED** [review]
If I recall correctly, Windows' (XP, 2003) ntldr still support the NOSERIALMICE switch too, which kind of does the same. It may be worth adding its support there too.

EDIT: See later comment.

**Comment by HBelusca** [comment]
Note for others: I'll merge this PR myself once it gets the approvals.

**Review by HBelusca — CHANGES_REQUESTED** [review]
It doesn't seem Windows FASTDETECT accept any extra parameter (like the =COMxx). Instead, FASTDETECT is a single switch that disables detection of all the serial mice at once.
The =COMxx support is done with the `NOSERIALMICE` (which indeed accepts both the colon or the equals separators). But seeing your code, it kinds of implement that already, so it should be a minimal change in order to support both these in a Windows-compatible manner.

File: boot/freeldr/freeldr/arch/arm/macharm.c
Comment (disean) [reply boot/freeldr/freeldr/arch/arm/macharm.c:97]: This needs to be fixed globally.

**Review by HBelusca — APPROVED** [review]

File: boot/freeldr/freeldr/arch/uefi/uefihw.c
Comment (DarkFire01) [inline boot/freeldr/freeldr/arch/uefi/uefihw.c:127]: Hmm should i be doing something here?

**Review by DarkFire01 — APPROVED** [review]

File: boot/freeldr/freeldr/arch/uefi/uefihw.c
Comment (disean) [reply boot/freeldr/freeldr/arch/uefi/uefihw.c:127]: I think this function should at least detect the PS/2 and display controllers. The `Options` parameter can be safely ignored in an UEFI environment, and that doesn't cause any problems.

**Review by binarymaster — APPROVED** [review]

**Pull request merged**
